src/events/ldap.js

```diff
@@ -0,0 +1,95 @@
+const uuid4 = require('uuid4');
+const serverlessEvent = require('../proto/event_pb.js');
+const utils = require('../utils.js');
+const tracer = require('../tracer.js');
+const errorCode = require('../proto/error_code_pb.js');
+const eventInterface = require('../event.js');
+const moduleUtils = require('./module_utils.js');
+
+/**
+ * Wraps the ldap.js bind command function with tracing
+ * @param {Function} bindFunction The wrapped bind function from ldap.js module
+ * @returns {Function} The wrapped function
+ */
+function bindWrapper(bindFunction) {
+    return function internalBindWrapper(dn, password, controls, callback) {
+        try {
+            utils.debugLog(`LDAP.js bind() wrapper - dn: ${dn}`);
+
+            const url = '';
+            const socketPath = '';
+            const timeout = '';
+            const connectTimeout = '';
+            const tlsOptions = '';
+            const idleTimeout = '';
+            const strictDN = '';
+
+            const resource = new serverlessEvent.Resource([
+                url,
+                'ldap',
+                'bind',
+            ]);
+            const startTime = Date.now();
+            const bindEvent = new serverlessEvent.Event([
+                `ldap-${uuid4()}`,
+                utils.createTimestampFromTime(startTime),
+                null,
+                'ldap',
+                0,
+                errorCode.ErrorCode.OK,
+            ]);
+            bindEvent.setResource(resource);
+            eventInterface.addToMetadata(bindEvent, {
+                'LDAP Client': {
+                    URL: url || '',
```

Comment:
if url = '' so why do you write URL || '' it seems useless

Reply:
It was a placeholder, I've implemented now the url from the client.